src/viur/core/skeleton.py

```diff
@@ -870,182 +870,183 @@ def toDB(cls, skelValues: SkeletonInstance, update_relations: bool = True, **kwa
 
                 # Lock hashes from bones that must have unique values
                 if bone.unique:
-                    # Check if the property is really unique
-                    newUniqueValues = bone.getUniquePropertyIndexValues(skel, key)
-                    for newLockValue in newUniqueValues:
-                        lockObj = db.Get(db.Key(f"{skel.kindName}_{key}_uniquePropertyIndex", newLockValue))
-                        if lockObj:
+                    # Remember old hashes for bones that must have an unique value
+                    old_unique_values = []
+
+                    if f"{bone_name}_uniqueIndexValue" in db_obj["viur"]:
+                        old_unique_values = db_obj["viur"][f"{bone_name}_uniqueIndexValue"]
+                    # Check if the property is unique
+                    new_unique_values = bone.getUniquePropertyIndexValues(skel, bone_name)
+                    new_lock_kind = f"{skel.kindName}_{bone_name}_uniquePropertyIndex"
+                    for new_lock_value in new_unique_values:
+                        new_lock_key = db.Key(new_lock_kind, new_lock_value)
+                        if lock_db_obj := db.Get(new_lock_key):
+
                             # There's already a lock for that value, check if we hold it
-                            if lockObj["references"] != dbObj.key.id_or_name:
+                            if lock_db_obj["references"] != db_obj.key.id_or_name:
                                 # This value has already been claimed, and not by us
                                 # TODO: Use a custom exception class which is catchable with an try/except
                                 raise ValueError(
-                                    f"The unique value {skelValues[key]!r} of bone {key!r} has been recently claimed!")
+                                    f"The unique value {skel[bone_name]!r} of bone {bone_name!r} "
+                                    f"has been recently claimed!")
                         else:
                             # This value is locked for the first time, create a new lock-object
-                            newLockObj = db.Entity(db.Key(f"{skel.kindName}_{key}_uniquePropertyIndex", newLockValue))
-                            newLockObj["references"] = dbObj.key.id_or_name
-                            db.Put(newLockObj)
-                        if newLockValue in oldUniqueValues:
-                            oldUniqueValues.remove(newLockValue)
-                    dbObj["viur"][f"{key}_uniqueIndexValue"] = newUniqueValues
+                            lock_obj = db.Entity(new_lock_key)
+                            lock_obj["references"] = db_obj.key.id_or_name
+                            db.Put(lock_obj)
+                        if new_lock_value in old_unique_values:
+                            old_unique_values.remove(new_lock_value)
+                    db_obj["viur"][f"{db_key}_uniqueIndexValue"] = new_unique_values
```

Comment:
Ahh, @phorward you were to fast. `bone_name` would be correct. _db_key_ was simply too straightforward

```suggestion
                    db_obj["viur"][f"{bone_name}_uniqueIndexValue"] = new_unique_values
```